### PR TITLE
First pass at implementing webpack

### PR DESCRIPTION
**Pull request by lindsaycarr**
(this should not conflict with https://github.com/USGS-VIZLAB/gage-conditions/pull/5)

This PR adds the some basic publish steps, two js functions and a js file to pull them together to create some mouse-able dots on a page using the dv_stats data, adds webpack to the mix, and adds/uses d3.v5 (all modules right now so final js is 1.2 MB...). I followed @mwernimont's example at https://github.com/USGS-VIZLAB/viz-scratch/tree/master/webpack_demo and this blog https://medium.com/@svinkle/getting-started-with-webpack-and-es6-modules-c465d053d988.

We are thinking about adding the node modules to s3 and having a fetch step. I have gitignored the node_modules folder because it is HUGE! @mwernimont plans to look into how to snag just the modules we need (see #9). For now, in order to build, you are going to need to run the following:
```
npm install webpack --save
npm install webpack-cli --save
npm install d3
```

Resulting viz:
gage_conditions_gif

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "version": "",
+    "description": "",
+    "scripts": {
+        "start": "webpack --mode development"
+    },
+    "dependencies": {
+        "webpack": "^4.12.0"
+    },
+    "devDependencies": {
+        "webpack-cli": "^3.0.8"
```
Comment (mwernimont) [inline package.json]: Its odd we have `webpack` as a dependency and `webpack-cli` as only a dev dependency.  We need to have both to have webpack work, so I feel a stronger solution would be to have the 2 packages in both the production dependencies and the dev dependencies.  Which is super easy to do as you are installing them with node.  Just simply run `npm i webpack webpack-cli --save --save-dev` or have them both in the production line only, I just don't like them split up as they currently are.

**Comment by mwernimont** [comment]
@lindsaycarr I do have a question, when you showed me you had webpack run at the end of the remake(), did you see the changes you made immediately after the build?  I'm just wondering since webpack recreates the JS each time into the bundle and vizlab is already looking for the old JS file at the same time if we have the correct ordering of things running.

File: package.json (hunk 1 above)
Comment (mwernimont) [reply package.json]: Not that it matters for this specific test project, just in the future let's make sure they are together.

**Comment by lindsaycarr** [comment]
I think the order is correct because the publish step which moves `dist/bundle.js` into target happens after the system call to run webpack (see https://github.com/USGS-VIZLAB/gage-conditions/pull/6/files#diff-d0eb8752b11a4299633d0c40e695cf09R5). I have noticed that I need to run `unpublish()` before `vizmake()` to see changes. I think if I had the individual js files as targets and make those dependencies of the webpack step, we might be able to avoid that.

File: package.json (hunk 1 above)
Comment (lindsaycarr) [reply package.json]: Tried to troubleshoot a little with marty. No luck, so we created #8 for now.

**Pull request merged**